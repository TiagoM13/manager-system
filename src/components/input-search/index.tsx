import React from 'react';

import { MagnifyingGlass } from '@phosphor-icons/react';

type InputSearchProps = React.ComponentProps<'input'> & {};

export const InputSearch: React.FC<InputSearchProps> = ({ ...rest }) => {
  return (
    <div className="flex items-center">
      <input
        {...rest}
        type="text"
        className="text-sm border-r-0 border border-slate-400 py-1.5 px-4 rounded-l-lg h-9 outline-sky-500 min-w-60"
      />

      <button
        disabled={rest.disabled}
        type="submit"
        className="bg-sky-600 outline-sky-500 hover:bg-sky-500 disabled:bg-sky-700 rounded-r-lg py-1 px-2.5 h-9"
      >
        <MagnifyingGlass weight="bold" color="white" className="size-5" />
      </button>
    </div>
  );
};
