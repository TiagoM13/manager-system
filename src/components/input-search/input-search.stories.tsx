import React from 'react';

import { Meta } from '@storybook/react';

import { InputSearch as Search } from '.';

export default {
  title: 'Components/InputSearch',
  component: Search,
  parameters: {
    layout: 'centered',
  },
} as Meta;

export const InputSearch = () => {
  return <Search placeholder="pesquisar..." />;
};
