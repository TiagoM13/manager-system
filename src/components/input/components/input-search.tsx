import React from 'react';
import { Control, Controller } from 'react-hook-form';

import { twMerge } from 'tailwind-merge';

import { MagnifyingGlass } from '@phosphor-icons/react';

type InputSearchProps = React.ComponentProps<'input'> & {
  name?: string;
  control?: Control;
  className?: string;
};

export const InputSearch: React.FC<InputSearchProps> = ({
  name,
  control,
  className,
  ...rest
}) => {
  return (
    <div className="flex items-center">
      {control ? (
        <Controller
          name={name || ''}
          control={control}
          render={({ field }) => (
            <input
              {...field}
              type="search"
              className={twMerge(
                `w-full text-sm border-r-0 border border-slate-400 py-1.5 px-4 rounded-l-lg h-[36px] outline-offset-0 outline-sky-500 min-w-60`,
                className,
              )}
              {...rest}
            />
          )}
        />
      ) : (
        <input
          type="search"
          className={twMerge(
            `w-full text-sm border-r-0 border border-slate-400 py-1.5 px-4 rounded-l-lg h-[36px] outline-offset-0 outline-sky-500`,
            className,
          )}
          {...rest}
        />
      )}

      <div className="flex items-center bg-sky-600 disabled:bg-sky-700 rounded-r-lg py-1 px-2.5 h-[36px]">
        <MagnifyingGlass
          data-testid="icon-search"
          weight="bold"
          color="white"
          className="size-5"
        />
      </div>
    </div>
  );
};
