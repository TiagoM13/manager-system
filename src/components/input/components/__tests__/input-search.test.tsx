import React from 'react';

import { fireEvent, render } from '@testing-library/react';

import { InputSearch } from '../input-search';

describe('<InputSearch/>', () => {
  test('should render correctly component', () => {
    const placeholder = 'Pesquisar aqui';

    const { getByPlaceholderText, getByTestId } = render(
      <InputSearch name="teste" placeholder={placeholder} />,
    );

    expect(getByTestId('icon-search')).toBeInTheDocument();
    expect(getByPlaceholderText(placeholder)).toBeInTheDocument();
  });

  test('should change the input search', () => {
    const placeholder = 'Pesquisar aqui';

    const handleChange = jest.fn();

    const { getByPlaceholderText, asFragment } = render(
      <InputSearch
        name="teste"
        placeholder={placeholder}
        onChange={handleChange}
      />,
    );

    const input = getByPlaceholderText(placeholder) as HTMLInputElement;

    fireEvent.change(input, { target: { value: 'Teste' } });

    expect(input.value).toBe('Teste');
    expect(handleChange).toHaveBeenCalled();

    expect(asFragment()).toMatchSnapshot();
  });
});
