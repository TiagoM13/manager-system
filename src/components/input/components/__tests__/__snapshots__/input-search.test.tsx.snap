// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputSearch/> should change the input search 1`] = `
<DocumentFragment>
  <div
    class="flex items-center"
  >
    <input
      class="w-full text-sm border border-slate-400 py-1.5 px-4 rounded-l-lg h-[36px] outline-offset-0 outline-sky-500 disabled:opacity-60"
      placeholder="Pesquisar aqui"
      type="search"
    />
    <button
      class="flex items-center bg-sky-600 disabled:bg-sky-700 outline-offset-0 outline-sky-500 rounded-r-lg py-1 px-2.5 h-[36px] hover:bg-sky-500"
      type="submit"
    >
      <svg
        class="size-5"
        data-testid="icon-search"
        fill="white"
        height="1em"
        viewBox="0 0 256 256"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M232.49,215.51,185,168a92.12,92.12,0,1,0-17,17l47.53,47.54a12,12,0,0,0,17-17ZM44,112a68,68,0,1,1,68,68A68.07,68.07,0,0,1,44,112Z"
        />
      </svg>
    </button>
  </div>
</DocumentFragment>
`;
